### PR TITLE
Moving meat types from snacks.dm to meat.dm

### PR DESCRIPTION
**Pull request by Probe1**
This is my first PR and the last two days of work comprise of my entire lifetime of coding.  Please review this thoroughly.
- All meat types are now in [meat.dm].  This includes only the basic meat types like raw chicken or bear meat.  It also includes eggs.
- 123 recipes that called on meat types not already in meat.dm repathed to now look in meat.dm (ie /obj/item/weapon/reagent_containers/food/snacks/egg  is now /obj/item/weapon/reagent_containers/food/snacks/meat/egg)
- Smartfridges now smartly take anything from meat.dm instead of just monkeymeat.
- Plant bags now pick up all meat types instead of just plain meat and eggs.
- xenomeat renamed from 'meat' to 'xeno meat'
- meatbread is now located below xenomeatbread and spidermeatbread to prevent it from overwriting them during cooking and turning xeno/spidermeatbread into regular meatbread.
- microwave.dm also reordered to carpmeat and eggs below meat instead of above.

addresses https://github.com/d3athrow/vgstation13/issues/4586

I'm not sure if this needs a changelog since it's mostly backend stuff but if it does I'll figure out how to do that.

### CODE REVIEW

**Comment by 9600bauds** [comment]
welcome
welcome to the coderbus

File: code/game/machinery/kitchen/microwave.dm [hunk 1]
```diff
@@ -238,7 +238,7 @@
 				display_name = "Turnovers"
 				items_measures[display_name] = "turnover"
 				items_measures_p[display_name] = "turnovers"
-			if (istype(O,/obj/item/weapon/reagent_containers/food/snacks/carpmeat))
+			if (istype(O,/obj/item/weapon/reagent_containers/food/snacks/meat/carpmeat))
```
Comment (ComicIronic) [inline code/game/machinery/kitchen/microwave.dm]: This should be above the slab of meat check.

**Comment by ComicIronic** [comment]
How many recipes have you tested with this?

**Comment by Probe1** [comment]
I tested three recipes from each category and visually inspected most of the recipes code to check for mistakes like turning eggplant into /meat/eggplant.  

After reading your reply I decided to go back and test more.  This PR introduces a bug that makes the recipe for spider and xenomeatbread loafs turn into regular meat loafs.  I have *no fucking idea* how.  I'll update when I find out.

**Comment by FudgePucker** [comment]
Wowzers, my first PR was just changing a vending machine.
Good job.

**Comment by Probe1** [comment]
Well I meant to push the first commit but I have no idea how or what the second one is.  I'm scared.

**Comment by Probe1** [comment]
31471f8 moves meatbread to the bottom of the meatbreads list since its the most basic form and prevents it from overwriting xenomeatbread and whatnot.  

@FudgePucker It's not so bad.  I moved 5 meat items from snacks.dm to meat.dm then replace all'd 200 recipes to have the new obj path.  It just looks big I swear.

**Comment by ComicIronic** [comment]
Meat organisation :+1:

**Comment by IratePirate** [comment]
>meat_migration

Not sure why this made me laugh.

**Pull request merged**